WordPress/src/main/java/org/wordpress/android/networking/menus/MenusRestWPCom.java

```diff
@@ -0,0 +1,162 @@
+package org.wordpress.android.networking.menus;
+
+import android.content.Context;
+import android.text.TextUtils;
+
+import com.android.volley.VolleyError;
+import com.wordpress.rest.RestRequest;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.wordpress.android.WordPress;
+import org.wordpress.android.models.MenuModel;
+import org.wordpress.android.util.AppLog;
+
+import java.net.HttpURLConnection;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.wordpress.android.util.VolleyUtils.statusCodeFromVolleyError;
+import static org.wordpress.android.networking.menus.MenusDataModeler.*;
+
+/**
+ * Interface for WordPress.com Menus REST API methods.
+ *
+ * ref: https://developer.wordpress.com/docs/api/
+ */
+public class MenusRestWPCom {
+    public interface IMenusDelegate {
+        Context getContext();
+        long getSiteId();
+        void onMenuReceived(int statusCode, MenuModel menu);
+        void onMenusReceived(int statusCode, List<MenuModel> menus);
+        void onMenuCreated(int statusCode, MenuModel menu);
+        void onMenuDeleted(int statusCode, MenuModel menu, boolean deleted);
+        void onMenuUpdated(int statusCode, MenuModel menu);
+    }
+
+    private static final String MENU_REST_PATH = "/sites/%s/menus/%s";
+    private static final String MENUS_REST_PATH = "/sites/%s/menus";
+    private static final String CREATE_MENU_REST_PATH = "/sites/%s/menus/new";
+    private static final String DELETE_MENU_REST_PATH = "/sites/%s/menus/%s/delete";
+
+    //
+    // JSON keys for fetchAllMenus response object
+    //
+    public static final String MENU_KEY = "menu";
+    public static final String ALL_MENUS_MENUS_KEY = "menus";
+
+    private static final String DELETED_KEY = "deleted";
+
+    private final IMenusDelegate mDelegate;
+
+    public MenusRestWPCom(IMenusDelegate delegate) {
+        if (delegate == null) {
+            throw new IllegalArgumentException("IMenusDelegate cannot be null");
+        }
+        mDelegate = delegate;
+    }
+
+    public boolean createMenu(final MenuModel menu) {
+        // a name string is required for the REST call
+        if (menu == null || TextUtils.isEmpty(menu.name)) return false;
+        String path = formatPath(CREATE_MENU_REST_PATH, null);
+        Map<String, String> params = new HashMap<>();
+        params.put(MENU_NAME_KEY, menu.name);
+        WordPress.getRestClientUtilsV1_1().post(path, params, null, new RestRequest.Listener() {
+            @Override public void onResponse(JSONObject response) {
+                menu.menuId = response.optLong(MENU_ID_KEY);
+                mDelegate.onMenuCreated(HttpURLConnection.HTTP_OK, menu);
```

Comment:
> I think I wouldn't send the HTTP status code down to the listener at all

The original intent was not to send this status through. After writing some tests I realized I needed to include some way to check for errors.

First thought was to add an `onError` method, but that presented its own problems (mainly due to lack of foresight). If there were multiple REST calls to respond to there was no way to tell which call threw the error since I neglected to include a way to identify individual calls.

I'm going to take another swing at the error handling:
- [ ] add an `onError` interface method to `MenusListener`
- [ ] return integer IDs for REST calls
- [ ] remove the `statusCode` parameter from the existing interface methods
- [ ] create an enum to describe known errors